File: leetCode/fukun/201-300/lc_220.js
var containsNearbyAlmostDuplicate = function (nums, k, t) {
    let tmp = {};
    for (let i = 0; i < nums.length; i++) {
        for (let j = -t; j <= t; j++) {
            if (tmp[nums[i] + j] !== undefined && (i - tmp[nums[i] + j] <= k)) return true;
        }
        tmp[nums[i]] = i;
    }
    return false;
};
// console.log(containsNearbyAlmostDuplicate([1, 5, 9, 1, 5, 9], 2, 3));
// console.log(containsNearbyAlmostDuplicate([1, 0, 1, 1], 1, 2));
console.log(containsNearbyAlmostDuplicate([1, 2, 3, 1], 3, 0));

File: leetCode/XSS/lc_220.java
public class lc_220 {
    public static void main(String[] args) {
        System.out.print(new Solution().containsNearbyAlmostDuplicate(new int[]{-1,2147483647},1,2147483647));
    }
    static class Solution {
        public boolean containsNearbyAlmostDuplicate(int[] nums, int k, int t) {
            if(nums.length<=1||k<=0)return false;
            int len=nums.length;
            for(int i=0;i<len;i++)
            {
                for(int j=1;j<=k&&i+j<len;j++)
                {

                    if((long)Math.abs((long)nums[i]-(long)nums[i+j])<=(long)t)
                    {
                        return true;
                    }
                }
            }
            return false;

        }
    }

}
